src/main/java/com/github/webicitybrowser/webicity/core/renderer/RendererContext.java

```diff
@@ -1,10 +1,14 @@
 package com.github.webicitybrowser.webicity.core.renderer;
 
+import com.github.webicitybrowser.spec.fetch.FetchEngine;
+import com.github.webicitybrowser.spec.url.URL;
 import com.github.webicitybrowser.webicity.core.AssetLoader;
 
 public interface RendererContext {
 
 	AssetLoader getAssetLoader();
+	FetchEngine getFetchEngine();
+	URL currentDocumentURL();

```

Comment:
Can you call this getCurrentDocumentURL()? Since the others start with get
Also, put a blank line between each method.